plugins/integration/providers/cielo24/lib/api/KalturaCielo24JobProviderData.php

```diff
@@ -0,0 +1,159 @@
+	 */
+	public $username;
+	
+	/**
+	 * Api key for service provider
+	 * @var string
+	 * @readonly
+	 */
+	public $password;
+	
+	/**
+	 * Transcript content language
+	 * @var KalturaLanguage
+	 */
+	public $spokenLanguage;
+	
+	/**
+	 * should replace remote media content
+	 * @var bool
+	 */
+	public $replaceMediaContent;
+	
+	private static $map_between_objects = array
+	(
+		"entryId",
+		"flavorAssetId",
+		"captionAssetFormats",
+		"priority",
+		"fidelity",
+		"username",
+		"password",
+		"spokenLanguage",
+		"replaceMediaContent",
+	);
+	
+	/* (non-PHPdoc)
+	 * @see KalturaObject::getMapBetweenObjects()
+	 */
+	public function getMapBetweenObjects ( )
+	{
+		return array_merge ( parent::getMapBetweenObjects() , self::$map_between_objects );
+	}
+	
+	public function validateForUsage($sourceObject, $propertiesToSkip = array())
+	{
+		$entryId = $this->entryId;
+		$entry = entryPeer::retrieveByPK($entryId);
+		if(!$entry)
+			throw new KalturaAPIException(KalturaErrors::INVALID_ENTRY_ID, $entryId);
+	
+		$flavorAssetId = $this->flavorAssetId;
+		if($flavorAssetId)
+		{
+			$flavorAsset = assetPeer::retrieveById($flavorAssetId);
+			if(!$flavorAsset || $flavorAsset->getEntryId() != $entryId)
+				throw new KalturaAPIException(KalturaErrors::FLAVOR_ASSET_ID_NOT_FOUND, $flavorAssetId);
+		}
+		
+		$cielo24ParamsMap = kConf::get('cielo24','integration');
+		$supportedLanguages = $cielo24ParamsMap['languages'];
+		if($this->spokenLanguage)
+		{
+			if (!isset($supportedLanguages[$this->spokenLanguage]))
+				throw new KalturaAPIException(KalturaCielo24Errors::LANGUAGE_NOT_SUPPORTED, $this->spokenLanguage);
+		}
+	
+		return parent::validateForUsage($sourceObject, $propertiesToSkip = array());
+	}
+	
+	public function toObject($object_to_fill = null, $props_to_skip = array())
+	{
+		$object = parent::toObject($object_to_fill, $props_to_skip);
+	
+		$entryId = $object->getEntryId();
+		$entry = entryPeer::retrieveByPK($entryId);
+		$partnerId = $entry->getPartnerId();
+	
+		$cielo24OptionsObj = Cielo24Plugin::getPartnerCielo24Options($partnerId);
+		$object->setUsername($cielo24OptionsObj->username);
+		$object->setPassword($cielo24OptionsObj->password);
+		
+		if(!$object->getFlavorAssetId())
+		{
+			$sourceAsset = assetPeer::retrieveOriginalReadyByEntryId($entryId);
+			if(!$sourceAsset)
+				throw new KalturaAPIException(KalturaCielo24Errors::NO_FLAVOR_ASSET_FOUND, $entryId);
+			$object->setFlavorAssetId($sourceAsset->getId());
+		}
+		
+		$cielo24ParamsMap = kConf::get('cielo24','integration');
+		if(!$object->getSpokenLanguage())
+		{
+			$object->setSpokenLanguage($cielo24ParamsMap['default_language']);
+		}
+	
+		$formatsString = $object->getCaptionAssetFormats();
+		if($formatsString)
+		{
+			$formatsArray = explode(',', $formatsString);
+			$excludedFormats = $cielo24ParamsMap['exclude_formats'];
+			$sanitizedFormatsArray = array();
+			foreach($formatsArray as $format)
+			{
+				$format = preg_replace("/[^A-Z]/", "", $format);
```

Comment:
@shpiNi for forward compatibility, I would add `_` as whitelisted characater as well